keps/sig-autoscaling/2702-graduate-hpa-api-to-GA/README.md

```diff
@@ -0,0 +1,511 @@
+a cloud provider API, or upon an external software-defined storage or network
+control plane.
+
+For each of these, fill in the following—thinking about running existing user workloads
+and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+The HPA v2 APIs allow users to configure multiple metrics, each with a separate target. A recommendation is calculated
+for each metric and the largest recommendation is used.  The more metrics are added to a given HPA the longer it will
+take to reconcile. The HPA is single-threaded processing recommendations one-at-a-time. When default reconciliation
+period is 15 seconds.  If there is too much work to do reconciliation will slow down and happen less frequently than
+every 15 seconds.  This will cause autoscaling to be less responsive at high scale.
+
+Previously v1 scaled along two dimensions, number of HPA and number of pods selected by each HPA (linearly).
+Now it will scale with the number of metrics defined in HPAs and the number of pods selected each metric (linearly).
+
+Additionally, v2 adds a behavior structure which allows the user configure that rate and delay of scaling and down.
+Enforcing these constraints require storing previous recommendations and scaling events in memory. The longer the
+configured interval the more memory is used. The maximum window allows is 60 minutes ([code](https://pkg.go.dev/k8s.io/api/autoscaling/v2beta2#HPAScalingRules))
+so 240 recommendations / events per configured metric. Each recommendation is an `int32` and `time.Time`.
+Each scaling event is an `int32`, a `time.Time` and a `bool` ([code](https://pkg.go.dev/k8s.io/api/autoscaling/v2beta2#HPAScalingRules))
+so the memory footprint is relatively small.
+It will scale linearly with the number of metrics defined and the size of the HPA's configured window.
+
+###### Will enabling / using this feature result in any new API calls?
+
+<!--
+Describe them, providing:
+  - API call type (e.g. PATCH pods)
+  - estimated throughput
+  - originating component(s) (e.g. Kubelet, Feature-X-controller)
+Focusing mostly on:
+  - components listing and/or watching resources they didn't before
+  - API calls that may be triggered by changes of some Kubernetes resources
+    (e.g. update of object X triggers new updates of object Y)
+  - periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)
+-->
+No
```

Comment:
Maybe say, "No, not in comparison to using the existing v2beta2 APIs, but of course using HPA results in new API calls as described above.".